### DISCUSSION TITLE
Copilot repeats comments... to infinity

### DISCUSSION THREAD

**Discussion by HatLink** [community/community#13583] [Copilot Conversations] [answered]
I'll write a comment that's about 4 lines and it'll either repeat that commend endlessly, make infinite comments without any text or actually write code.

I've only tried JavaScript so far so this may be an issue with the way it handles JavaScript.

Still ily github for making this great ai programmer,
Copilot is great!

**Comment by CharlonTank** [comment, marked as answer]
First: read this: https://beta.openai.com/docs/guides/code/introduction
You should try having fun with GPT3 OpenAI playground https://beta.openai.com/playground to understand how it works and how to talk to the AI.
Repeating a thing multiple times is usual and happens a lot. For example: https://news.ycombinator.com/item?id=26442211.

You have to give the AI a good beginning. For example:

If you ask to complete `1+1` it could totally go crazy and do

`1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+`

But if you input `1+1=`
There is a high chance it will complete with just a `2`

Also, give your AI some context by adding comments on top of what you want the AI to complete.

See this AI as an augmentation of your brain, instead of a magic thing.

If you want the AI to create a big chunk of code, it will mostly fail. But if you have a function to do that you can cut in 5 functions, name these functions and code one of them, comment on top of each function what it needs to do, and try to use copilot to fill the gap.

You can also think more about concepts and explanations more, no-coding, just words, and let the AI do the coding, but if you're not clear enough in your head, the AI will mostly not help you.
